src/lib/connect/ali.ts

```diff
@@ -68,17 +68,29 @@ function bindEventHandler() {
 		stream.socketReady()
 	})
 
-	my.onSocketMessage((res) => {
+	my.onSocketMessage((res: any) => {
 		if (typeof res.data === 'string') {
 			const buffer = Buffer.from(res.data, 'base64')
 			proxy.push(buffer)
 		} else {
 			const reader = new FileReader()
 			reader.addEventListener('load', () => {
-				let data = reader.result
-
-				if (data instanceof ArrayBuffer) data = Buffer.from(data)
-				else data = Buffer.from(data, 'utf8')
+				let data:
+					| string
+					| ArrayBuffer
+					| Buffer<ArrayBufferLike>
+					| null = reader.result
+
+				// @TODO: @robertsLando We need to handle this situation as if data is null, Buffer.from(null) will throw an Error
+				if (data === null) {
+					throw new Error('')
+				}
+
+				if (data instanceof ArrayBuffer) {
+					data = Buffer.from(data)
+				} else {
+					data = Buffer.from(data, 'utf8')
+				}
```

Comment:
Here again a strict TypeScript configuration reveals a potential error that will be uncaught and thus bubble back to our end users code. I left this `new Error('')` blank for now since I still need to understand better what to do in this situation. I'm not completely familiar with this code base.

Reply:
No worries, every code base has problems, even if you have 100% coverage, 100% stryker score and 0 SonarQube issue you have problems that are not clearly identified. I'm just highlighting the potential benefits of a stricter configuration.
This is actually really positive as we can improve this part since we know it has a potential issue here. 👍